### PR TITLE
PC-2705 Add tests for OmniReturnsClient and Client

### PR DESCRIPTION
**Pull request by trang-mai-nguyen**

### CODE REVIEW

File: spec/seko_ecom_api/clients/omni_returns_client_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    context 'success' do
+      let(:seko_response) { { "CarrierId": 123, 'CarrierName': 'foo', 'Errors': [] } }
+      it do
+        shipment = subject.create_shipment({})
+        expect(shipment).to be_a SekoEcomAPI::Shipment
+      end
+    end
+
+    context 'unsuccess' do
```
Comment (andychongyz) [inline spec/seko_ecom_api/clients/omni_returns_client_spec.rb]:
```suggestion
    context 'failure' do
```

Comment (andychongyz) [reply spec/seko_ecom_api/clients/omni_returns_client_spec.rb]: And we can also test the one where the body is some random HTML string

**Review by andychongyz — CHANGES_REQUESTED** [review]

File: lib/seko_ecom_api/clients/omni_returns_client.rb [hunk 2]
```diff
@@ -48,10 +50,13 @@ def handle_response(response)
...
     end
 
+    # need to be manual parsed here instead of in the middleware
+    # since Seko response's Content-Type is always in 'text/html'
+    # with the response body can be in JSON or in html
     def parse_response(response)
       parsed_response = JSON.parse(response.body)
       parsed_response.deep_transform_keys(&:underscore)
-    rescue JSON::ParserError
+    rescue JSON::ParserError, NoMethodError
```
Comment (andychongyz) [inline lib/seko_ecom_api/clients/omni_returns_client.rb]: Why do we have to rescue NoMethodError here?

Comment (trang-mai-nguyen) [reply lib/seko_ecom_api/clients/omni_returns_client.rb]: Oh I forgot to remove it

Comment (andychongyz) [reply lib/seko_ecom_api/clients/omni_returns_client.rb]: @trang-mai-nguyen But you haven't remove it lol

Comment (andychongyz) [inline lib/seko_ecom_api/clients/omni_returns_client.rb]:
```suggestion
    rescue JSON::ParserError
```

**Review by andychongyz — CHANGES_REQUESTED** [review]

File: lib/seko_ecom_api/clients/omni_returns_client.rb (hunk 2 above)
Comment (trang-mai-nguyen) [reply lib/seko_ecom_api/clients/omni_returns_client.rb]: @andychongyz omg what the heck lol. Removed it

**Review by andychongyz — APPROVED** [review]

**Pull request merged**